pkg/harness/controller/controller.go

```diff
@@ -78,6 +79,14 @@ func (in *stackRunController) Start(ctx context.Context) (retErr error) {
 	return in.postStart(retErr)
 }
 
+func (in *stackRunController) Finish(stackRunErr error) error {
```

Comment:
It could also be fixed by wrapping all early returns from the `Start` function with `postStart`

Reply:
I feel like handling this a level up is less brittle to devs forgetting to add the postStart wrapper

Reply:
Restored